sly.el

```diff
@@ -4407,21 +4407,17 @@ in Lisp when committed with \\[sly-edit-value-commit]."
   "Hook run by `sly-change-directory'.
 The functions are called with the new (absolute) directory.")
 
-(defun sly-change-directory (directory)
+(defun sly-change-directory (directory &optional evaluator)
   "Make DIRECTORY become Lisp's current directory.
 Return whatever slynk:set-default-directory returns."
   (let ((dir (expand-file-name directory)))
-    (prog1 (sly-eval `(slynk:set-default-directory
-                       ,(sly-to-lisp-filename dir)))
+    (prog1 (funcall (or evaluator 'sly-eval)
+                    `(slynk:set-default-directory
+                      (slynk-backend:filename-to-pathname
```

Comment:
It's just this one line that fixes it: `slynk-backend:filename-to-pathname`.
But this only works if the other cd functions call `sly-change-directory`.